src/IRI.php

```diff
@@ -301,12 +301,10 @@ protected function parse_iri(string $iri)
             if ($match[1] === '') {
```

Comment:
We could simplify this further by passing [PREG_UNMATCHED_AS_NULL](https://www.php.net/manual/en/pcre.constants.php#constant.preg-unmatched-as-null) to `preg_match` (available since PHP 7.2).